src/components/token/client-authentication/validate-private-key-jwt.ts

```
@@ -0,0 +1,274 @@
+  }
+  const header = decodeProtectedHeader(clientAssertion);
+  const signature = jwtParts[2];
+
+  if (!signature || signature.trim().length === 0) {
+    throw new ParseTokenRequestError(
+      "Invalid client_assertion JWT: The signature must not be empty"
+    );
+  }
+
+  if (!payload.sub) {
+    throw new ParseTokenRequestError(
+      "Invalid client_assertion JWT: Missing subject in client JWT assertion"
+    );
+  }
+
+  if (!payload.iss) {
+    throw new ParseTokenRequestError(
+      "Invalid client_assertion JWT: Missing issuer in client JWT assertion"
+    );
+  }
+
+  if (!payload.aud) {
+    throw new ParseTokenRequestError(
+      "Invalid client_assertion JWT: Missing audience in client JWT assertion"
+    );
+  }
+
+  if (!payload.exp || typeof payload.exp !== "number") {
+    throw new ParseTokenRequestError(
+      "Invalid client_assertion JWT: Missing or invalid expiry in client JWT assertion"
+    );
+  }
+
+  if (payload.nbf) {
+    //We don't care about this claim but it can be included and
+    //can cause a ParseError if the claim is invalid
+    logger.warn(
+      "nbf claim included in client JWT assertion, this claim is not needed: https://docs.sign-in.service.gov.uk/integrate-with-integration-environment/authenticate-your-user/#create-a-jwt"
+    );
+
+    if (typeof payload.nbf !== "number") {
+      throw new ParseTokenRequestError(
+        "Invalid client_assertion JWT: Invalid nbf claim in client JWT assertion"
+      );
+    }
+  }
+
+  if (!payload.iat) {
+    logger.warn(
+      "iat claim not included in client JWT assertion, this claim is recommended: https://docs.sign-in.service.gov.uk/integrate-with-integration-environment/authenticate-your-user/#create-a-jwt"
+    );
+  }
+
+  if (payload.iat && typeof payload.iat !== "number") {
+    throw new ParseTokenRequestError(
+      "Invalid client_assertion JWT: Invalid iat claim in client JWT assertion"
+    );
+  }
+
+  if (!payload.jti) {
+    logger.warn(
```

does this not need to throw an error if it is required?

---

So this matches our actual implementation - we say this field is required but don't actually [validate it]( https://github.com/govuk-one-login/authentication-api/blob/7cdc5162ecc44f3b5488e5a88ca829879c4bb3e6/orchestration-shared/src/main/java/uk/gov/di/orchestration/shared/validation/PrivateKeyJwtClientAuthValidator.java#L41) (yet!). This is probably because we're reserving the right to do validation/_other stuff_ with this field in the future